File: 프로그래머스/unrated/133499. 옹알이 （2）/옹알이 （2）.py
def solution(babbling):
    answer = 0
    word = ["aya", "ye", "woo", "ma"]
    for bb_idx in range(len(babbling)) :
        prev = ""
        w_idx = 0
        while w_idx < 4 :
            if prev != word[w_idx] and babbling[bb_idx][0 : len(word[w_idx])] == word[w_idx] :
                babbling[bb_idx] = babbling[bb_idx][len(word[w_idx]) :]
                prev = word[w_idx]
                w_idx = -1
            w_idx += 1
        if len(babbling[bb_idx]) == 0 :
            answer += 1
    return answer

File: 프로그래머스/unrated/133499. 옹알이 （2）/옹알이 （2）.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<string> babbling)
{
    int answer = 0;
    for (int bb_idx = 0; bb_idx < babbling.size(); ++bb_idx) {
        vector<string> word{ "aya", "ye", "woo", "ma" };
        string prev{ "" };
        for (int w_idx = 0; w_idx < word.size(); ++w_idx) {
            if (prev != word[w_idx] && babbling[bb_idx].substr(0, word[w_idx].length()) == word[w_idx]) {   
                babbling[bb_idx] = babbling[bb_idx].substr(word[w_idx].length(), babbling[bb_idx].length() - 1);
                prev = word[w_idx];
                w_idx = -1;
            }
        }
        if (babbling[bb_idx].empty()) ++answer;
    }
    return answer;
}